common/src/main/java/com/spredfast/kafka/connect/s3/S3.java

```diff
@@ -1,35 +1,27 @@
 package com.spredfast.kafka.connect.s3;
 
-import com.amazonaws.auth.AWSStaticCredentialsProvider;
-import com.amazonaws.auth.BasicAWSCredentials;
-import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration;
 import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.S3ClientOptions;
 import java.util.Map;
+import java.util.Objects;
 
 public class S3 {
 
   public static AmazonS3 s3client(Map<String, String> config) {
-    boolean s3PathStyle = Boolean.parseBoolean(config.get("s3.path_style"));
-    String s3Endpoint = config.get("s3.endpoint");
+    // Use default credentials provider that looks in Env + Java properties + profile + instance
+    // role
+    AmazonS3 s3Client = new AmazonS3Client();
 
-    AmazonS3 s3Client;
-    if (s3Endpoint == null || "".equals(s3Endpoint)) {
-      s3Client = AmazonS3ClientBuilder.standard().withPathStyleAccessEnabled(s3PathStyle).build();
-    } else {
-      // Testing with localstack
-      s3Client =
-          AmazonS3ClientBuilder.standard()
-              .withEndpointConfiguration(
-                  new EndpointConfiguration(config.get("s3.endpoint"), config.get("region")))
-              .withCredentials(
-                  new AWSStaticCredentialsProvider(
-                      new BasicAWSCredentials(
-                          config.get("aws.accessKeyId"), config.get("aws.secretAccessKey"))))
-              .withPathStyleAccessEnabled(s3PathStyle)
-              .build();
+    // If worker config sets explicit endpoint override (e.g. for testing) use that
+    String s3Endpoint = config.get("s3.endpoint");
+    if (s3Endpoint != null && !Objects.equals(s3Endpoint, "")) {
```

Comment:
I wouldn't ignore the empty string. If the default endpoint should be used, the parameter should be completely omitted.

Reply:
I just rolled back this version of the file to it's previous working production version, but this makes sense to me. I'll update.